### DISCUSSION TITLE
Link is not processed

### DISCUSSION THREAD

**Discussion by rozkomnadzor** [BeanieODM/beanie#612] [Question] [answered]
Model cannot identify and make fetch for the reference 
`questions[].answers[].image`
is there a solution for this without changing the structure? do I need to use aggregation?
![image](https://github.com/roman-right/beanie/assets/43275267/2cfe41d9-7d44-4295-bba2-a19c31fabb66)

**Comment by roman-right** [comment, marked as answer]
Hi! Yes, non-top-level links are not supported yet. But you can fetch them manually using the 'fetch' method of the link or yes, you can use lookup aggregations.

I plan to support such links too. It is a bit tricky to detect them, but I think it is doable.

**Reply by Ray-Escobar** [reply]
I think it would be really great addition to the library since it would make it very easy to handle complex objects. Is it a feature that would be very far-fetched? I assume it goes quite deep into reflection of the object.

In your opinion then, the best way to go around it would be to use the fetch function of the object? 

Would I then have to create the links manually before saving them? (because I know that with links within nested objects the object is not actually turned into link, but instead the full objects is stored)

or if i call fetch on the object, would it be updated correctly using the object id? Trying to work around the best way possible to handle links in nested objects when saving and fetching.

Thanks a lot.
